### DISCUSSION TITLE
Which OSM tags used when factoring height, width, length, weight in truck profile routing

### DISCUSSION THREAD

**Discussion by answerquest** [valhalla/valhalla#4924] [Q&A] [answered]
Hi, the docs at https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/ page don't elaborate on exactly **which** osm tags on a highway are used by valhalla to figure out if a road should be taken or not if we have truck costing with height, width, length, weight numbers given in `costing_options.truck` as shown here: https://github.com/valhalla/valhalla/discussions/3295#discussioncomment-1244921

I've been trying to find out the same but haven't been able to locate a clear answer yet. Would anybody know?

**Comment by kevinkreiser** [comment, marked as answer]
i would have a look at taginfo: https://github.com/valhalla/valhalla/blob/master/taginfo.json

it lists all the tags we care about so its exhaustive but if you can manage to ignore the stuff you dont care about the information you are looking for will be in there. for a more specific look though, here is the relevant code: https://github.com/valhalla/valhalla/blob/a01d80ce737083a733ab203c44fbadb3642fe9b7/lua/graph.lua#L1817-L1823

**Reply by answerquest** [reply]
thanks, this is very helpful!
